src/components/pages/course-form/CourseFormContainer.jsx

```diff
@@ -1,11 +1,102 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { Form, Input, Space, Button } from 'antd';
+import { ModuleForm, ModuleCard } from '../module-form';
+import styled from 'styled-components';
+
+const StyledSpace = styled(Space)`
+  &&& {
+    width: 100%;
+  }
+`;
+
+// CourseFormContainer — No specific parent, props isn't coming down quite yet
+export default props => {
+  const [form] = Form.useForm();
+  const [modalVisible, setModalVisible] = useState(false);
+
+  const showModuleModal = () => setModalVisible(true);
+  const hideModuleModal = () => setModalVisible(false);
+
+  const onFinish = values => {
+    // we will do something different here once
+    // we're ready to hit endpoints!!
+    console.log({ values });
+  };
+
+  const onModuleAdd = newModule => {
+    const existingModules = form.getFieldValue('modules') || [];
```

Comment:
I like your use of this pattern here and it's a great way to catch for empty datasets so that no errors are thrown! I need to be better about incorporating this pattern into my work in the future as well